src/com/perl5/lang/perl/idea/run/PerlRunProfileState.java

```diff
@@ -155,4 +163,54 @@ protected ProcessHandler startProcess() throws ExecutionException
 		return runProfile.getEnvs();
 	}
 
+	public List<String> getPerlLibraries(@Nullable VirtualFile scriptFile)
+	{
+		if (!PlatformUtils.isIntelliJ())
+		{
+			return Collections.emptyList();
+		}
+		else
+		{
+			PerlConfiguration runProfile = (PerlConfiguration) getEnvironment().getRunProfile();
+
+			// try to get used SDK
+			Sdk sdk;
+			if (runProfile.isUseAlternativeSdk())
+			{
+				sdk = ProjectJdkTable.getInstance().findJdk(runProfile.getAlternativeSdkPath());
+			}
+			else
+			{
+				Module moduleForFile = scriptFile == null ? null : ModuleUtilCore.findModuleForFile(scriptFile, getEnvironment().getProject());
+
+				sdk = PerlRunUtil.getModuleSdk(moduleForFile);
+				if (sdk == null)
+				{
+					// try project SDK
+					sdk = ProjectRootManager.getInstance(getEnvironment().getProject()).getProjectSdk();
+				}
+			}
+
+			if (sdk != null && sdk.getSdkType() == PerlSdkType.getInstance())
+			{
+				List<String> incPaths = PerlSdkType.getInstance().getINCPaths(sdk.getHomePath());
+				List<String> list = new ArrayList<>();
+
+				for (VirtualFile file : sdk.getRootProvider().getFiles(OrderRootType.CLASSES))
```

Comment:
I think we have a different point of view what this "libraries" are.

In our case this libraries are part of the SDK and should be used for each project (eg some of the cpan modules are stored in a separat directory). I think it is not very useful to configure it manually for each project.

So if I add this additional libraries to the SDK (that is shared between projects) and I can use the same configuration in multiple projects it is more useful. (I think the PyCharm plugin has the same functionality)